scripts/irods/test/test_iadmin.py

```diff
@@ -1823,6 +1823,41 @@ def test_iadmin_rmuser_removes_associated_acls__issue_7778(self):
             self.admin.run_icommand(["irmdir", test_collection_name])
             self.admin.run_icommand(["irm", "-f", test_dataobj_name])
 
+    def test_modzone_does_not_accept_invalid_zone_names__issue_7722(self):
+        # Show that "iadmin modzone name" no longer accepts invalid zone names.
+        invalid_zone_name = 'überZone'
+        ec, _, _ = self.admin.assert_icommand(
+            ['iadmin', 'modzone', self.admin.zone_name, 'name', invalid_zone_name],
+            'STDERR',
+            ['-130000 SYS_INVALID_INPUT_PARAM'],
+            input='yes')
+        self.assertNotEqual(ec, 0)
+
+        # Show the zone name was not updated.
+        ec, out, err = self.admin.assert_icommand(['iadmin', 'lz'], 'STDOUT')
+        self.assertEqual(ec, 0)
```

Comment:
Consider using `desired_rc` in the `assert_icommand(...)` function.

The logging provided by the function may help catch issues later.

Reply:
See https://github.com/irods/irods/pull/8174/files#r1946983500.

Reply:
This check is actually redundant.

Will remove it.

Reply:
Correct.

Reply:
Removed assertion.